runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,454 @@
+  else
+    for _, line in iter_lines(bufnr, 1, 5) do
+      -- Always ignore case
+      line = line:lower()
+      if line:find("vb_name") or line:find("begin vb%.form") or line:find("begin vb%.mdiform") or line:find("begin vb%.usercontrol") then
+        vim.bo[bufnr].filetype = "vb"
+      else
+        vim.bo[bufnr].filetype = "form"
+      end
+    end
+  end
+end
+
+function M.fs(path, bufnr)
+
+end
+
+function M.header(bufnr)
+  for _, line in iter_lines(bufnr, 1, 200) do
+    if line:find("^@interface") or line:find("^@end") or line:find("^@class") then
+      if vim.g.c_syntax_for_h then
+        vim.bo[bufnr].filetype = "objc"
+      else
+        vim.bo[bufnr].filetype = "objcpp"
+      end
+      return
+    end
+  end
+  if vim.g.c_syntax_for_h then
+      vim.bo[bufnr].filetype = "c"
+  elseif vim.g.ch_syntax_for_h then
+      vim.bo[bufnr].filetype = "ch"
+  else
+      vim.bo[bufnr].filetype = "cpp"
+  end
+end
+
+function M.idl(bufnr)
+  for _, line in iter_lines(bufnr, 1, 50) do
+    -- Always ignore case
+    line = line:lower()
+    if line:find('^%s*import%s+"unknwn"%.idl') or line:find('^%s*import%s+"objidl"%.idl') then
+      vim.bo[bufnr].filetype = "msidl"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "idl"
+end
+
+function M.inc(path, bufnr)
+
+end
+
+function M.inp(bufnr)
+  if get_lines(bufnr, 1):find("^%*") then
+    vim.bo[bufnr].filetype = "abaqus"
+  else
+    for _, line in iter_lines(bufnr, 1, 500) do
+      if line:find("header surface data") then
+        vim.bo[bufnr].filetype = "trasys"
+        return
+      end
+    end
+  end
+end
+
+function M.is_rapid(path, bufnr)
+
+end
+
+function M.lpc(path, bufnr)
+
+end
+
+function M.lprolog(path, bufnr)
+
+end
+
+function M.m(path, bufnr)
+
+end
+
+-- Rely on the file to start with a comment.
+-- MS message text files use ';', Sendmail files use '#' or 'dnl'
+function M.mc(bufnr)
+  for _, line in iter_lines(bufnr, 1, 20) do
+    if line:find("^%s*#") or line:find("^%s*[dD][nN][lL]") then
+      -- Sendmail .mc file
+      vim.bo[bufnr].filetype = "m4"
+      return
+    elseif line:find("^%s*;") then
+      vim.bo[bufnr].filetype = "msmessages"
+      return
+    end
+  end
+  -- Default: Sendmail .mc file
+  vim.bo[bufnr].filetype = "m4"
+end
+
+function M.mm(path, bufnr)
+
+end
+
+function M.mms(bufnr)
+  for _, line in iter_lines(bufnr, 1, 20) do
+    if line:find("^%s*%%") or line:find("^%s*//") or line:find("^%*") then
+      vim.bo[bufnr].filetype = "mmix"
+      return
+    elseif line:find("^%s*#") then
+      vim.bo[bufnr].filetype = "make"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "mmix"
+end
+
+function M.mod(path, bufnr)
+
+end
+
+-- This function checks if one of the first five lines start with a dot. In
+-- that case it is probably an nroff file.
+function M.nroff(bufnr)
+  for _, line in iter_lines(bufnr, 1, 5) do
+    if line:find("^%.") then
+      vim.bo[bufnr].filetype = "nroff"
```

Comment:
This check is used elsewhere, so we need to mirror the return value (`1` if `nroff`, `0` else).

(Same for Perl below, when that gets ported.)

Reply:
(Alternative -- forward looking -- would be to return `nroff` already and then just check `if detect.nroff() then ... end` at the calling site in `filetype.lua.)